app/src/main/java/com/example/songme/data/repository/TrackRepository.kt

```diff
@@ -0,0 +1,29 @@
+package com.example.songme.data.repository
+
+import com.example.songme.data.source.OnTracksLoadedCallback
+import com.example.songme.data.source.TrackDataSource
+import com.example.songme.data.source.remote.TrackRemoteDataSource
+
+class TrackRepository(private val dataSource: TrackRemoteDataSource) : TrackDataSource.Remote {
+
+    companion object {
+        var single_instance: TrackRepository? = null
+        fun getInstance(): TrackRepository? {
+            if (single_instance == null)
+                single_instance = TrackRepository(TrackRemoteDataSource.getInstance()!!)
+            return single_instance
+        }
+    }
```

Comment:
companion object cho xuống cuối cùng của class 
https://kotlinlang.org/docs/reference/coding-conventions.html#class-layout
Dùng elvis operator để thay thế check null 
```suggestion
    companion object {
        private var instance: TrackRepository? = null
        fun getInstance(): TrackRepository? {
            if (single_instance == null)
                single_instance = TrackRepository(TrackRemoteDataSource.getInstance()!!)
            return single_instance
        }
    }
```